lib/ansible/modules/storage/dellemc/dellemc_pmax_createsg.py

```diff
@@ -0,0 +1,200 @@
+    required: true
+  sgname:
+    description:
+      - "Storage Group name 32 Characters no special characters other than
+      underscore"
+    required: true
+  slo:
+    description:
+      - "Service Level for the storage group, Supported on VMAX3 and All Flash
+      and PowerMAX NVMe Arrays running PowerMAX OS 5978 and above.  Default is
+      set to Diamond, but user can override this by setting a different value."
+    required: false
+  srp_id:
+    description:
+      - "Storage Resource Pool Name, Default is set to SRP_1, if your system
+      has mainframe or multiple pools you can set this to a different value to
+      match your environment"
+    required: false
+  unispherehost:
+    description:
+      - "Fully Qualified Domain Name or IP address of Unisphere for PowerMax
+      host."
+    required: true
+  universion:
+    description:
+      - "Integer, version of unipshere software  e.g. 90"
+    required: true
+  verifycert:
+    description:
+      - "Boolean, security check on ssl certificates"
+    type: bool
+    required: true
+  vol_size:
+    description:
+      - "Integer value for the size of volumes.  All volumes will be created
+      with same size.  Use dellemc_pmax_addvol to add additional volumes if you
+      require different sized volumes once storage group is created."
+    required: true
+  volumeIdentifier:
+    description:
+      - "String up to 64 Characters no special character other than
+      underscore sets a label to make volumes easily identified on hosts can
+      run Dell EMC inq utility command to see this label is  inq -identifier
+      device_name"
+    required: false
+  user:
+    description:
+      - "Unisphere username"
+  password:
+    description:
+      - "password for Unisphere user"
+  workload:
+    description:
+      - "Block workload type, optional and can only be set on VMAX3 Hybrid
+      Storage Arrays. Default None."
+    required: false
+requirements:
+  - Ansible
+  - "Unisphere for PowerMax version 9.0 or higher."
+  - "VMAX All Flash, VMAX3, or PowerMax storage Array."
+  - "PyU4V version 3.0.0.8 or higher using PIP python -m pip install PyU4V"
+'''
+EXAMPLES = '''
+---
+- name: "Provision Storage For DB Cluster"
+  connection: local
+  hosts: localhost
+  vars:
+    array_id: 000197600156
+    password: smc
+    sgname: Ansible_SG
+    unispherehost: "192.168.1.123"
+    universion: "90"
+    user: smc
+    verifycert: false
+
+  tasks:
+    - name: "Create New Storage Group and add data volumes"
+      dellemc_pmax_createsg:
+        array_id: "{{array_id}}"
+        cap_unit: GB
+        num_vols: 1
+        password: "{{password}}"
+        sgname: "{{sgname}}"
+        slo: Diamond
+        srp_id: SRP_1
+        unispherehost: "{{unispherehost}}"
+        universion: "{{universion}}"
+        user: "{{user}}"
+        verifycert: "{{verifycert}}"
+        vol_size: 1
+        workload: None
+        volumeIdentifier: 'REDO'
+'''
+RETURN = '''
```

Comment:
RETURN is optional, but useful end user to understand what module returns.

Reply:
I will change module to return a dictionary describing the storage group that has been created.

Reply:
added sample return for successful run
dellemc_pmax_createsg:
    description: Information about storage group created
    returned: success 
    type: dict
    sample: '{
        "storagegroup_detail": {
            "VPSaved": "100.0%",
            "base_slo_name": "Diamond",
            "cap_gb": 1.0,
            "compression": true,
            "device_emulation": "FBA",
            "num_of_child_sgs": 0,
            "num_of_masking_views": 0,
            "num_of_parent_sgs": 0,
            "num_of_snapshots": 0,
            "num_of_vols": 1,
            "service_level": "Diamond",
            "slo": "Diamond",
            "slo_compliance": "STABLE",
            "srp": "SRP_1",
            "storageGroupId": "Ansible_SG",
            "type": "Standalone",
            "unprotected": true,
            "vp_saved_percent": 100.0
            }
        }'
'''